staging/src/k8s.io/client-go/dynamic/dynamicinformer/informer_test.go

```diff
@@ -248,6 +248,72 @@ func TestDynamicSharedInformerFactory(t *testing.T) {
 	}
 }
 
+func TestStoppableDynamicSharedInformerFactory(t *testing.T) {
+	timeout := time.Duration(3 * time.Second)
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	onListErrorFunc := func(error) bool {
+		return true
+	}
+	testObject := newUnstructured("extensions/v1beta1", "Deployment", "ns-foo", "name-foo")
+	fakeClient := fake.NewSimpleDynamicClient(runtime.NewScheme(), []runtime.Object{testObject}...)
+	target := dynamicinformer.NewDynamicSharedInformerFactoryWithOptions(fakeClient, 0, dynamicinformer.WithOnListError(onListErrorFunc))
+
+	gvr := schema.GroupVersionResource{Group: "extensions", Version: "v1beta1", Resource: "deployments"}
+	// retrieve the informer for the resource forces the factory to create the informer.
+	_ = target.ForResource(gvr)
+	infCtx, infCancel := context.WithCancel(ctx)
+	target.Start(infCtx.Done())
+	doneChannel, ok := target.DoneChannelFor(gvr)
+	if !ok {
+		t.Errorf("Unable to retrieve done channel for gvr")
+	}
+	go infCancel()
+
+	select {
+	case <-doneChannel:
+		break
+	case <-ctx.Done():
+		t.Errorf("informer has not stopped itself, waited %v", timeout)
+		break
+	}
+}
+
+// TODO: Get this working such that you're stopping the specific informer instead of the whole factory
+// TODO: Clean up fake dynamic client hacks.
+func TestSpecificInformerStopOnListError(t *testing.T) {
+	timeout := time.Duration(3 * time.Second)
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	onListErrorFunc := func(error) bool {
+		// when this is true it passes, when false it does not
+		// This is expected.
+		return true
+	}
+	testObject := newUnstructured("extensions/v1beta1", "Deployment", "ns-foo", "name-foo")
+	fakeClient := fake.NewSimpleDynamicClient(runtime.NewScheme(), []runtime.Object{testObject}...)
+	fakeClient.ForceListError()
```

Comment:
```suggestion
	fakeClient.AddReactor("list", "deployment", testing.ReactionFunc(func(...) {
	   // something, something is actually important here
	   return something, something, fmt.Errorf("oh no!")
	})
```

(note, this might not be exactly the right syntax -- the reactors are pretty brittle)

https://pkg.go.dev/k8s.io/client-go/testing#ReactionFunc
https://pkg.go.dev/k8s.io/client-go/testing#Fake.AddReactor

Reply:
Fixed